config/initializers/highlight_override.rb

```diff
@@ -90,9 +99,23 @@ def query(user_uuid:)
       end
 
       highlights.sort_by!{ |highlight| [source_id_order[highlight.source_id], highlight.order_in_source] }
+    else
+      # Have to sort by something for pagination to be sensible, choose created_at
+      highlights.sort_by!(&:created_at)
     end
 
-    highlights
+    local_page = page || Api::V0::Swagger::Models::Highlight::DEFAULT_HIGHLIGHTS_PAGE
+    local_per_page = per_page || Api::V0::Swagger::Models::Highlight::DEFAULT_HIGHLIGHTS_PER_PAGE
+
+    first_index_in_page = (local_page - 1) * local_per_page
```

Comment:
we could use will_paginate on ruby results too (not just models).  that could help sweep away the paging logic out of the model code.

Reply:
good idea.